backend/src/test/kotlin/org/darkSolace/muse/userModule/UserServiceTests.kt

```diff
@@ -74,117 +75,211 @@ class UserServiceTests {
         val user3 = userService.createUser(User(username = "testUser9", password = "123", email = "test9@example.com"))
-        Assertions.assertEquals(count + 3, userRepository.count())
+
+        // get all users
         val all = userService.getAll()
+        //should be three more than before
+        Assertions.assertEquals(count.toInt() + 3, all.count())
+        //check if all created test users are there
         Assertions.assertTrue(all.contains(user))
         Assertions.assertTrue(all.contains(user2))
         Assertions.assertTrue(all.contains(user3))
     }
 
     @Test
     fun getAllWithRole() {
+        // create test user
         userService.createUser(User(username = "testUser10", password = "123", email = "test10@example.com"))
-        val members = userService.getAllWithRole(Role.MEMBER)
         userService.createUser(
             User(
-                username = "testUser11",
-                password = "123",
-                email = "test11@example.com",
+                username = "testUser11", password = "123", email = "test11@example.com",
                 role = Role.ADMINISTRATOR
             )
         )
-        val admins = userService.getAllWithRole(Role.ADMINISTRATOR)
         userService.createUser(
             User(
-                username = "testUser12",
-                password = "123",
-                email = "test12@example.com",
+                username = "testUser12", password = "123", email = "test12@example.com",
                 role = Role.MODERATOR
             )
         )
-        val mods = userService.getAllWithRole(Role.MODERATOR)
         userService.createUser(
             User(
-                username = "testUser13",
-                password = "123",
-                email = "test13@example.com",
+                username = "testUser13", password = "123", email = "test13@example.com",
                 role = Role.SUSPENDED
             )
         )
-        val suspended = userService.getAllWithRole(Role.SUSPENDED)
+
+        //get all users by role
         val count = userRepository.count()
+        val members = userService.getAllWithRole(Role.MEMBER)
+        val admins = userService.getAllWithRole(Role.ADMINISTRATOR)
+        val mods = userService.getAllWithRole(Role.MODERATOR)
+        val suspended = userService.getAllWithRole(Role.SUSPENDED)
 
+        //check if everything adds up and all users per group have the specified role
         Assertions.assertEquals(count.toInt(), members.count() + admins.count() + mods.count() + suspended.count())
         Assertions.assertTrue(members.all { it.role == Role.MEMBER })
         Assertions.assertTrue(mods.all { it.role == Role.MODERATOR })
         Assertions.assertTrue(admins.all { it.role == Role.ADMINISTRATOR })
         Assertions.assertTrue(suspended.all { it.role == Role.SUSPENDED })
     }
 
+    @Test
+    fun getAllWithUserTags() {
+        // create test users
```

Comment:
Please clean up the previously created users for this tests to be able to assert definitive counts at the end. Or save the previous counts and assert the new additions. As you prefer.

Also please add a test user which has multiple tags and is thus added to all counts per tags it has but not for tags is doesn't have. If that makes any sense.